src/utils/promise-to-iterable.ts

```diff
@@ -0,0 +1,32 @@
+import { $$asyncIterator } from 'iterall';
+
+export function createIterableFromPromise<T>(promise: Promise<T>): AsyncIterator<T> {
+  let isResolved = false;
+
+  return {
```

Comment:
The `return` of `AsyncIterator` should not return a value, only `done=false`.
If setting `done=true` while the Promise is resolved, the value ignored and then the iterator is done.